README.md

```diff
@@ -6,8 +6,130 @@
+here are:
+
+- `Autoloader` the ZF autoloader, used to load the following services and storage backends 
+- `Service` the main service class used to interact with the underlying features of the library
+- `Factories` the `UserStorage`, `UserSecretStorage` and `StateStorage` are creating the specified storage backend based 
+   on the way you configured Tiqr
+
+## Creating the Service
+An end to end example on how to configure, create and work with one of the libraries classes.
+
+### Config
+To create the Tiqr Service, we encourage you to create a factory in your consuming code base. In order to create the 
+service, you will first need to create a Tiqr server configuration. A brief example is listed below. Note that this
+example is taken from a Symfony parameters file. This YAML is later transformed to a PHP Array. But for readability the
+YAML format is retained in this example:
+
+```yaml
+ Options for the tiqr library
+    tiqr_library_options:
+        general:
+            identifier: 'tiqr.example.com'
+            # The name of the service that uses tiqr for second factor authentication.
+            # Used by both the TiqrService and the legacy Tiqr_Service during enrollment and when sending an
+            # authentication notification.
+            name: 'tiqr'
+            auth_protocol: 'tiqrauth'
+            enroll_protocol: 'tiqrenroll'
+            ocra_suite: 'OCRA-1:HOTP-SHA1-6:QH10-S'
+            logoUrl: '%base_url%/images/header-logo.png'
+            infoUrl: 'https://tiqr.example.com/tiqr_info_url'
+        library:
+          apns:
+            certificate: 'files/apns.pem'
+            environment: production
+        accountblocking:
+            maxAttempts: 5
+            # temporarilyBlockDuration:
+            # maxTemporarilyBlocks:
+        storage:
+            statestorage:
+              type: 'memcache'
+              arguments:
+                prefix: 'tiqr_'
+                servers:
+                  - host: '192.168.66.11'
+            userstorage:
+              type: 'pdo'
+              arguments:
+                table: 'user'
+                dsn: 'mysql:host=192.168.66.11;dbname=tiqr'
+                username: 'tiqr'
+                password: 'super_secret'
+            usersecretstorage:
+              type: 'pdo'
+              arguments:
+                table: 'tiqrusersecret'
+                dsn: 'mysql:host=192.168.66.11;dbname=tiqr-user-secret'
+                username: 'tiqr'
+                password: 'super_secret_2'
+```
+
+### Creation
+Creating the Tiqr_Service is now as simple as creating a new instance with the configuration
+
+```php
+# Convert the Yaml to a PHP array see: https://github.com/OpenConext/Stepup-tiqr/blob/1aa2b686cd4dbffc0d96135042875a881a0e232e/src/Tiqr/TiqrConfiguration.php
+$options = $this->getTiqrConfigurationOptions();
+# Create the Tiqr_Service
+$service = new Tiqr_Service($options)
+```
+
+### Usage
```

Comment:
Thinking about it, what is useful to add here is the way that the library is intented to be used:
- Tiqr_Service is created bu user of the library. Tiqr_Service creates Tiqr_StateStorage and Tiqr_DeviceStorage from configuration you need to provide.
- session_id is intended for getting (polling) enroll and authentication status in the web browser
- Tiqr_UserStorage, UserSecretStorage and Tiqr_Message are factories that you need to call yourself

Reply:
I changed the usage example. Adding more pseudo code and focus it more on an authentication scenario.